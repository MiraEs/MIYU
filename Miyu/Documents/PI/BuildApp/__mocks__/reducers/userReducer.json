{
  "user": {
    "notificationCount": 0,
    "status": "AUTH_STATUS_AUTHENTICATED",
    "isGuest": false,
    "username": "geoff.lawson@build.com",
    "shippingAddresses": [],
    "billingAddresses": [],
    "defaultShippingAddressId": null,
    "storeCredit": 0,
    "rep": {},
    "tierInfo": {},
    "customerId": 3984706,
    "password": null,
    "firstName": "Geoff",
    "lastName": "Lawson",
    "email": "geoff.lawson@build.com",
    "userName": "JeffWithG",
    "isPro": false,
    "isSubscriber": null,
    "apiUser": {
      "userType": "CUSTOMER",
      "name": "Geoff Lawson",
      "title": null,
      "avatar": null,
      "firstName": null,
      "lastName": null
    },
    "pricebookId": null
  },
  "notificationSettings": {
    "MARKETING": {
      "PUSH": true,
      "SMS": false
    }
  },
  "userPushNotificationsEnabled": true,
  "isCreatingUser": false,
  "isLoading": false,
  "isLoggingIn": false,
  "isLoggingInSocial": false,
  "isLoggedIn": true,
  "_error": "",
  "errors": {
    "signUp": "",
    "login": "",
    "rep": "",
    "update": ""
  },
  "deviceToken": "",
  "impersonatorId": null
}