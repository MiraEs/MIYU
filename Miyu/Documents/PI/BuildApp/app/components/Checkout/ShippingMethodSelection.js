import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	StyleSheet,
	View,
} from 'react-native';
import {
	Text,
} from 'BuildLibrary';
import EventEmitter from '../../lib/eventEmitter';
import styles from '../../lib/styles';
import PhoneHelper from '../../lib/PhoneHelper';
import OptionSelectButton from '../OptionSelectButton';
import ShippingMethod from '../ShippingMethod';


const componentStyles = StyleSheet.create({
	indent: {
		marginHorizontal: styles.measurements.gridSpace1,
	},
});

class ShippingMethodSelection extends Component {

	getAvailableShippingOptions = (shippingOptions) => {
		return shippingOptions ? shippingOptions.filter((option) => option.customerAvailable) : [];
	};

	renderShippingMethod = (shippingOptions, selectedShippingIndex) => {
		if (shippingOptions) {
			return <ShippingMethod shippingMethod={shippingOptions[selectedShippingIndex]} />;
		}
	};

	/*
		bounce is a stub so OptionSelectButton.bounce can be called
	 */
	bounce = () => {};// NOSONAR

	render() {
		const { cart, isDisabled, hasError, onPress, user } = this.props;
		const { shippingOptions } = cart.cart;
		const { selectedShippingIndex } = cart;
		const filtered = this.getAvailableShippingOptions(shippingOptions);
		const phone = PhoneHelper.getPhoneNumberByUserType(user);

		if (shippingOptions && (shippingOptions.length !== filtered.length)) {
			return (
				<OptionSelectButton
					style={componentStyles.indent}
					onPress={() => EventEmitter.emit('onCallUs', phone, true)}
				>
					<View style={styles.elements.flex}>
						<Text weight="bold">{shippingOptions[0].shippingOption}</Text>
						<Text color="error">
							Due to higher shipping charges for this area, a shipping quote is required
							{' '}before placing an order. Please contact us at {PhoneHelper.formatPhoneNumber(phone)} to place your order.
						</Text>
					</View>
				</OptionSelectButton>
			);
		}

		return (
			<OptionSelectButton
				key="shippingMethod"
				onPress={onPress}
				text="Select a shipping method"
				ref={(ref) => this.shippingMethod = ref}
				style={componentStyles.indent}
				isDisabled={isDisabled}
				hasError={hasError}
				errorMessage="Please finish filling in your information before submitting your order."
				accessibilityLabel="Select Shipping Method Button"
			>
				{this.renderShippingMethod(shippingOptions, selectedShippingIndex)}
			</OptionSelectButton>
		);
	}
}

ShippingMethodSelection.propTypes = {
	cart: PropTypes.object.isRequired,
	isDisabled: PropTypes.bool,
	hasError: PropTypes.bool,
	onPress: PropTypes.func.isRequired,
	user: PropTypes.object.isRequired,
};

ShippingMethodSelection.defaultProps = {
	isDisabled: false,
	hasError: false,
};

export default ShippingMethodSelection;
