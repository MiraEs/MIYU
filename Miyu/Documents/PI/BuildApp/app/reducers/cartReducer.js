import {
	GET_SESSION_CART_SUCCESS,
	UPDATE_SESSION_CART_SUCCESS,
	DELETE_SESSION_CART_SUCCESS,
	DELETE_SESSION_CART_ITEM_SUCCESS,
	GET_CUSTOMER_CARTS_SUCCESS,
	SET_SESSION_CART_ITEM_DELETE_STATUS,
	CLEAR_SESSION_CART_ITEM_DELETE_STATUS,
	SET_SESSION_CART_ITEM_PROPS,
	SET_SELECTED_SHIPPING_INDEX,
	ADD_COUPON_SUCCESS,
	REMOVE_COUPON_SUCCESS,
	CLEAR_SESSION_CART_SUCCESS,
	LOADING_SESSION_CART,
	UNDO_STATUS,
	COPY_SESSION_CART_SUCCESS,
	LOAD_QUOTE_SUCCESS,
	UPDATE_CART_ITEM_BOUNCE,
	GET_DELIVERY_DATES,
	GET_DELIVERY_DATES_SUCCESS,
	SET_DELIVERY_DATE,
	GET_SESSION_CART_ERRORS,
	SET_SESSION_CART_ERRORS,
} from '../constants/CartConstants';
export const initialState = {
	cart: {
		sessionCartItems: [],
		sessionCartErrors: [],
		checkingErrors: false,
	},
	carts: [],
	selectedShippingIndex: 0,
	sessionCartItemDeleteQueue: [],
	quoteId: null,
	isLoading: false,
	bounceFirstRowOnMount: false,
};

export function updateCartItem(cartItem, cartItems) {

	return cartItems.map((currentItem) => {
		let item = { ...currentItem };

		if (item.itemKey === cartItem.itemKey) {
			item = Object.assign({}, item, cartItem);
			item.hasChanged = true;
		}

		if (item.hasSubItems) {
			item.subItems = item.subItems.map((subItem) => {
				if (subItem.itemKey === cartItem.itemKey) {
					subItem = Object.assign({}, subItem, cartItem);
					item.hasChanged = true;
				}
				return subItem;
			});
		}
		return item;
	});
}

function cartReducer(state = initialState, action = {}) {
	const { cartItem, deleteId, itemKey } = action.payload || {};
	let cart, sessionCartItems;

	switch (action.type) {
		case COPY_SESSION_CART_SUCCESS:
			cart = {
				...action.payload,
				sessionCartItems: action.payload.sessionCartItems || [],
			};
			return {
				...state,
				isLoading: false,
				cart,
			};
		case GET_SESSION_CART_SUCCESS:
			sessionCartItems = action.cart.sessionCartItems || [];

			state.sessionCartItemDeleteQueue.forEach((queueItem) => {
				sessionCartItems.forEach((cartItem) => {
					if (cartItem.hasSubItems) {
						cartItem.subItems.forEach((subItem) => {
							if (subItem.itemKey === queueItem.itemKey) {
								subItem.deleteStatus = UNDO_STATUS;
							}
						});
					}

					if (cartItem.itemKey === queueItem.itemKey) {
						cartItem.deleteStatus = UNDO_STATUS;
					}
				});
			});

			cart = {
				...action.cart,
				sessionCartItems,
			};

			return {
				...state,
				isLoading: false,
				cart,
			};
		case UPDATE_SESSION_CART_SUCCESS:
			cart = {
				...action.cart,
				sessionCartItems: action.cart.sessionCartItems || [],
			};

			return {
				...state,
				isLoading: false,
				selectedShippingIndex: !action.zipCode ? 0 : state.selectedShippingIndex,
				cart,
			};
		case DELETE_SESSION_CART_SUCCESS:
			return {
				...state,
				isLoading: false,
				cart: {
					sessionCartItems: [],
				},
			};
		case DELETE_SESSION_CART_ITEM_SUCCESS:
			return {
				...state,
				isLoading: false,
				sessionCartItemDeleteQueue: state.sessionCartItemDeleteQueue.filter((item) => item.itemKey !== itemKey),
			};
		case GET_CUSTOMER_CARTS_SUCCESS:
			return {
				...state,
				isLoading: false,
				carts: action.carts.sort((a, b) => {
					if (a.created < b.created) {
						return 1;
					}
					return -1;
				}),
			};
		case SET_SESSION_CART_ITEM_DELETE_STATUS:
			cartItem.deleteStatus = UNDO_STATUS;
			cart = {
				...state.cart,
				sessionCartItems: updateCartItem(cartItem, state.cart.sessionCartItems),
			};

			return {
				...state,
				isLoading: false,
				sessionCartItemDeleteQueue: state.sessionCartItemDeleteQueue.concat({
					itemKey: cartItem.itemKey,
					status: UNDO_STATUS,
					deleteId,
				}),
				cart,
			};
		case CLEAR_SESSION_CART_ITEM_DELETE_STATUS:
			delete cartItem.deleteStatus;
			cart = {
				...state.cart,
				sessionCartItems: updateCartItem(cartItem, state.cart.sessionCartItems),
			};

			return {
				...state,
				isLoading: false,
				sessionCartItemDeleteQueue: state.sessionCartItemDeleteQueue.filter((item) => {
					if (item.itemKey === cartItem.itemKey) {
						clearTimeout(item.deleteId);
						return false;
					}
					return true;
				}),
				cart,
			};
		case SET_SESSION_CART_ITEM_PROPS:
			cart = {
				...state.cart,
				sessionCartItems: state.cart.sessionCartItems.map((item) => {
					if (item.itemKey === action.cartItem.itemKey) {
						item = Object.assign({}, item, action.props);
					}
					return item;
				}),
			};

			return {
				...state,
				isLoading: false,
				cart,
			};
		case SET_SELECTED_SHIPPING_INDEX:
			return {
				...state,
				isLoading: false,
				selectedShippingIndex: action.selectedShippingIndex > -1 ? action.selectedShippingIndex : 0,
			};
		case ADD_COUPON_SUCCESS:
			return {
				...state,
				isLoading: false,
				cart: action.cart,
			};
		case REMOVE_COUPON_SUCCESS:
			return {
				...state,
				isLoading: false,
				cart: action.payload,
			};
		case CLEAR_SESSION_CART_SUCCESS:
			return {
				...initialState,
			};
		case LOADING_SESSION_CART:
			return {
				...state,
				isLoading: true,
			};
		case LOAD_QUOTE_SUCCESS:
			return {
				...state,
				quoteId: action.quoteId,
			};
		case UPDATE_CART_ITEM_BOUNCE:
			return {
				...state,
				bounceFirstRowOnMount: action.payload.shouldBounce,
			};
		case GET_DELIVERY_DATES:
			delete state.cart.availableDeliveryDates;
			return { ...state };
		case GET_DELIVERY_DATES_SUCCESS:
			return {
				...state,
				cart: {
					...state.cart,
					availableDeliveryDates: action.payload,
				},
			};
		case SET_DELIVERY_DATE:
			return {
				...state,
				cart: {
					...state.cart,
					...action.payload,
				},
			};
		case GET_SESSION_CART_ERRORS: {
			return {
				...state,
				cart: {
					...state.cart,
					checkingErrors: true,
				},
			};
		}
		case SET_SESSION_CART_ERRORS: {
			return {
				...state,
				cart: {
					...state.cart,
					sessionCartErrors: action.payload,
					checkingErrors: false,
				},
			};
		}
		default:
			return {
				...state,
				isLoading: false,
			};
	}
}

export default cartReducer;
