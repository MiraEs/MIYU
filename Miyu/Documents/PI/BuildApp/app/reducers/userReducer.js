'use strict';

import {
	CREATE_USER_FAIL,
	UPDATE_USER_FAIL,
	REQUEST_CREATE_USER,
	AUTH_STATUS_UNKNOWN,
	AUTH_STATUS_ANONYMOUS,
	AUTH_STATUS_AUTHENTICATED,
	GET_CUSTOMER_ID_SUCCESS,
	SIGN_USER_OUT,
	GET_CUSTOMER_FAIL,
	GET_CUSTOMER_SUCCESS,
	GET_CUSTOMER_REP_SUCCESS,
	GET_CUSTOMER_REP_FAIL,
	GET_CUSTOMER_REWARDS_TIER,
	UPDATE_SIGNUP_ERROR,
	LOGIN_USER_ERROR,
	LOGIN_USER_SUCCESS,
	LOGIN_USER_BEGIN,
	LOGIN_SOCIAL_USER_BEGIN,
	LOGIN_USER_FINISH,
	LOGIN_USER_RESET,
	UPDATE_USER,
	UPDATE_DEVICE_TOKEN,
	KEYCHAIN_FETCH_CREDS_FAIL,
	SET_IMPERSONATOR_ID,
} from '../constants/Auth';
import {
	GET_NOTIFICATION_SETTINGS_SUCCESS,
	SAVE_NOTIFICATION_SETTINGS_SUCCESS,
	SAVE_NOTIFICATION_SETTINGS_FAIL,
	ALLOW_NOTIFICATIONS_FAILED,
	ALLOW_NOTIFICATIONS_SUCCESS,
} from '../constants/Notifications';
import {
	SOCIAL_GET_CLIENT_DATA,
	SOCIAL_LOGIN_PROMPT_CREDS,
	SOCIAL_LINK_USER,
	SOCIAL_LINK_USER_ERROR,
} from '../constants/SocialConstants';
import {
	SHOULD_PRESENT_NO_EMAIL_ERROR_MESSAGE,
} from '../constants/constants';

const ANONYMOUS_USER = {
	status: AUTH_STATUS_ANONYMOUS,
	isGuest: true,
};

const initialState = {
	user: {
		notificationCount: 0,
		status: AUTH_STATUS_UNKNOWN,
		isGuest: true,
		username: '',
		shippingAddresses: [],
		billingAddresses: [],
		defaultShippingAddressId: null,
		storeCredit: 0,
		rep: {},
		tierInfo: {},
	},
	notificationSettings: {
		MARKETING: {
			PUSH: true,
			SMS: false,
		},
	},
	userPushNotificationsEnabled: true,
	isCreatingUser: false,
	isLoading: false,
	isLoggingIn: true, // defaulting this to true so we have a chance to log in before data load
	isLoggingInSocial: true, // defaulting this to true so we have a chance to log in before data load
	isLoggedIn: false,
	_error: '',
	errors: {
		signUp: '',
		login: '',
		rep: '',
		update: '',
	},
	deviceToken: '',
	impersonatorId: null,
};

// eslint-disable-next-line complexity
const userReducer = (state = initialState, action = {}) => {
	function toggleNotification (type, enabled) {
		const notificationSettings = {...state.notificationSettings};
		notificationSettings[type] = notificationSettings[type] || {};
		notificationSettings[type].PUSH = enabled;
		return notificationSettings;
	}
	switch (action.type) {
		case UPDATE_SIGNUP_ERROR:
			return {
				...state,
				errors: {
					...state.errors,
					signUp: action.message,
				},
			};

		case KEYCHAIN_FETCH_CREDS_FAIL:
			return {
				...state,
				user: {
					...state.user,
					status: AUTH_STATUS_ANONYMOUS,
				},
			};

		case GET_CUSTOMER_ID_SUCCESS:
			return {
				...state,
				user: {
					...state.user,
					status: AUTH_STATUS_AUTHENTICATED,
					customerId: action.payload.customerId,
				},
				isLoggedIn: true,
			};

		case GET_CUSTOMER_SUCCESS:
			return {
				...state,
				user: {
					...state.user,
					...action.payload,
				},
			};

		case GET_CUSTOMER_FAIL:
			return {
				...state,
				user: ANONYMOUS_USER,
			};

		case GET_CUSTOMER_REP_SUCCESS:
			return {
				...state,
				user: {
					...state.user,
					rep: action.payload,
				},
			};

		case GET_CUSTOMER_REP_FAIL:
			return {
				...state,
				user: {
					...state.user,
					rep: {},
				},
				errors: {
					...state.errors,
					rep: action.error,
				},
			};

		case GET_CUSTOMER_REWARDS_TIER:
			return {
				...state,
				user: {
					...state.user,
					tierInfo: action.payload,
				},
			};

		case LOGIN_USER_BEGIN:
			return {
				...state,
				isLoggingIn: true,
				errors: {
					...state.errors,
					login: '',
				},
			};

		case LOGIN_SOCIAL_USER_BEGIN:
			return {
				...state,
				isLoggingInSocial: true,
				errors: {
					...state.errors,
					login: '',
				},
			};

		case LOGIN_USER_FINISH:
			return {
				...state,
				isLoggingIn: false,
				isLoggingInSocial: false,
			};

		case LOGIN_USER_RESET:
			return {
				...state,
				isLoggingIn: false,
				isLoggingInSocial: false,
				isLoggedIn: false,
				errors: {
					...state.errors,
					login: '',
				},
			};

		case LOGIN_USER_SUCCESS:
			return {
				...state,
				user: {
					...state.user,
					notificationCount: state.user.notificationCount,
					status: AUTH_STATUS_AUTHENTICATED,
					username: action.username,
					...action.payload,
				},
				isCreatingUser: false,
				isLoggedIn: true,
				errors: {
					...state.errors,
					login: '',
				},
			};

		case LOGIN_USER_ERROR:
			return {
				...state,
				user: {
					status: AUTH_STATUS_ANONYMOUS,
				},
				errors: {
					...state.errors,
					login: action.error,
				},
				impersonatorId: null,
			};

		case REQUEST_CREATE_USER:
			return {
				...state,
				isCreatingUser: true,
			};

		case UPDATE_DEVICE_TOKEN:
			return {
				...state,
				deviceToken: action.token,
			};

		case CREATE_USER_FAIL:
			return {
				...state,
				errors: {
					...state.errors,
					signUp: action.error,
				},
				isCreatingUser: false,
			};

		case UPDATE_USER_FAIL:
			return {
				...state,
				errors: {
					...state.errors,
					update: action.error,
				},
			};

		case SOCIAL_GET_CLIENT_DATA:
			return {
				...state,
				user: {
					...state.user,
					...action.data,
				},
			};

		case SOCIAL_LOGIN_PROMPT_CREDS:
			const { email, socialUserAccessToken } = action.data;
			return {
				...state,
				user: {
					...state.user,
					email,
					socialUserAccessToken,
				},
			};

		case SOCIAL_LINK_USER:
			return {
				...state,
				user: {
					...state.user,
					socialLinkError: '',
				},
			};

		case SOCIAL_LINK_USER_ERROR:
			return {
				...state,
				user: {
					...state.user,
					socialLinkError: action.error.message,
				},
			};

		case SIGN_USER_OUT:
			return {
				...state,
				user: {
					...ANONYMOUS_USER,
				},
				isLoggedIn: false,
				impersonatorId: null,
				notificationSettings: initialState.notificationSettings,
				userPushNotificationsEnabled: initialState.userPushNotificationsEnabled,
			};

		case GET_NOTIFICATION_SETTINGS_SUCCESS:
			return {
				...state,
				notificationSettings: action.payload.eventTypeNotifications,
				userPushNotificationsEnabled: action.payload.userPushNotificationsEnabled,
			};

		case SAVE_NOTIFICATION_SETTINGS_SUCCESS:
		case SAVE_NOTIFICATION_SETTINGS_FAIL:
			let payload = action.payload;
			if (!Array.isArray(payload)) {
				payload = [payload];
			}
			let notificationSettings = {...state.notificationSettings};
			payload.forEach(({ type, enabled }) => {
				notificationSettings = toggleNotification(type, enabled);
			});
			return {
				...state,
				notificationSettings,
			};
		case ALLOW_NOTIFICATIONS_FAILED:
		case ALLOW_NOTIFICATIONS_SUCCESS:
			return {
				...state,
				...action.payload,
			};
		case SHOULD_PRESENT_NO_EMAIL_ERROR_MESSAGE:
			return {
				...state,
				user: {
					...state.user,
					noEmailErrorMessage: action.present ? 'We were unable to create your account because the Facebook login you used is not associated with an email address. Please create a new account here.' : '',
				},
			};
		case UPDATE_USER:
			return {
				...state,
				user: {
					...state.user,
					...action.payload.user,
				},
			};
		case SET_IMPERSONATOR_ID:
			return {
				...state,
				impersonatorId: action.payload,
			};
		default:
			return state;
	}

};


export default userReducer;
