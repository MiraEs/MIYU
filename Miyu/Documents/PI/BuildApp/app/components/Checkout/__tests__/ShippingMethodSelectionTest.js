jest.mock('../../../../app/store/configStore', () => ({}));
jest.mock('BuildLibrary');
jest.mock('BuildNative');

jest.unmock('react-native');

import React from 'react';
import { shallow } from 'enzyme';
import ShippingMethodSelection from '../ShippingMethodSelection';
import cartReducer from '../../../../__mocks__/reducers/cartReducer';
import userReducer from '../../../../__mocks__/reducers/userReducer';


const defaultProps = {
	cart: cartReducer,
	isDisabled: false,
	hasError: false,
	onPress: jest.fn(),
	user: userReducer.user,
};

describe('ShippingMethodSelection component', () => {
	const wrapper = shallow(<ShippingMethodSelection {...defaultProps} />);

	beforeEach(() => jest.resetModules());

	it('should render with default props', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
