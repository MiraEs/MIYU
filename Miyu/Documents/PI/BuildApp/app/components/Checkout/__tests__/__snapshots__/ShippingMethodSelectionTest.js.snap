// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingMethodSelection component should render with default props 1`] = `
<OptionSelectButton
  accessibilityLabel="Select Shipping Method Button"
  boldText=""
  errorMessage="Please finish filling in your information before submitting your order."
  hasError={false}
  hideIcon={false}
  isConfigured={false}
  isDisabled={false}
  isSelected={false}
  onPress={[Function]}
  style={
    Object {
      "marginHorizontal": 8,
    }
  }
  text="Select a shipping method"
>
  <ShippingMethod
    shippingMethod={
      Object {
        "customerAvailable": true,
        "leadTime": null,
        "shippingCost": 0,
        "shippingName": "Standard Delivery (5 - 7 Business Days)",
        "shippingOption": "Standard Delivery",
        "shippingOptionId": 12,
      }
    }
  />
</OptionSelectButton>
`;
